model/multimap.go

```diff
@@ -0,0 +1,69 @@
+package model
+
+import (
+	"hash/maphash"
+	"strconv"
+)
+
+// Identifier needs to be implemented by any type to be used with MultiMap. The
+// value returned by ID() needs to be unique for every instance of Identifier.
+type Identifier interface {
+	ID() string
+}
+
+// MultiMap is a map with an n-dimensional index.
+type MultiMap[T any, T2 Identifier] interface {
+	Get(identifiers ...T2) T
+	Length() int
```

Comment:
```suggestion
	// Get an element from the MultiMap, given an n-dimensional index or
	// adds it by using the create function if it hasn't been created previously.
	Get(identifiers ...T2) T
	// Length of the elements in the MultiMap.
	Length() int
```

Plz add godocs to the methods of the interface, as opposed to adding them to the particular implementation of the MultiMap interface via the multiMap struct. What this achieves is showing up via intellisense when typing in an editor and using the method of the interface.